src/main/java/core/basesyntax/parser/Parser.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.parser;
+
+import core.basesyntax.model.Fruit;
+
+import java.util.List;
+import java.util.Set;
+
+public interface Parser {
+    List<Fruit> parseLines(List<String[]> lines, Set<String> types);
+}
```

Comment:
```suggestion
public interface Parser {
    List<FruitRecord> parseLines(List<String> lines);
}
```
instead of passing types, let's create a public enum with the types that will be available all across the program

I suggest to create a model `FruitRecord` to hold the information about each operation: 
`OperationType`, `Fruit` and `amount`

you will later be able to pass this `List<FruitRecord>` into your `storageService` to save them to storage